commit b772d1410a83034f7452c7c22658ba89d22c5b55
Author: Mike Lothian <mike@fireburn.co.uk>
Date:   Fri Dec 15 00:54:11 2017 +0000

    Revert "drm/amd/display: Release cached atomic state in S3."
    
    This reverts commit 0a214e2fb6b0a56519b6d5efab4b21475c233ee0.

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index f71fe6d2ddda..48a20042e051 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -598,7 +598,6 @@ static int dm_suspend(void *handle)
 
 	amdgpu_dm_irq_suspend(adev);
 
-	WARN_ON(adev->dm.cached_state);
 	adev->dm.cached_state = drm_atomic_helper_suspend(adev->ddev);
 
 	dc_set_power_state(dm->dc, DC_ACPI_CM_POWER_STATE_D3);
@@ -716,8 +715,6 @@ int amdgpu_dm_display_resume(struct amdgpu_device *adev)
 
 	ret = drm_atomic_helper_resume(ddev, adev->dm.cached_state);
 
-	adev->dm.cached_state = NULL;
-
 	amdgpu_dm_irq_resume_late(adev);
 
 	return ret;
